### PR TITLE
WIP: Replace uses of BusyBox with Agnhost

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We are continuing to move towards reducing the number of images
required to run the e2e test suite. Part of that is centralizing
images into the Agnhost and leveraging that when possible.

**Which issue(s) this PR fixes**:
Related to #76342

**Special notes for your reviewer**:
First pass at this; unsure how tests will do.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

/sig testing
/area e2e-test-framework
/priority important-soon

### CODE REVIEW

File: test/e2e_node/util.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ var kubeletAddress = flag.String("kubelet-address", "http://127.0.0.1:10255", "H
 
 var startServices = flag.Bool("start-services", true, "If true, start local node services")
 var stopServices = flag.Bool("stop-services", true, "If true, stop local node services after running tests")
-var busyboxImage = imageutils.GetE2EImage(imageutils.BusyBox)
+var busyboxImage = imageutils.GetE2EImage(imageutils.Agnhost)
```
Comment (saiyan86) [inline test/e2e_node/util.go]: We should just call this agnhostImage

**Comment by neolit123** [comment]
/lgtm
/approve
/assign @andrewsykim 

cc @bclau

File: test/e2e_node/util.go (hunk 1 above)
Comment (spiffxp) [reply test/e2e_node/util.go]: agree with @saiyan86, not touching this var makes for a much smaller diff, but agnhost is based off of alpine so the var name is now misleading

**Comment by spiffxp** [comment]
a glance at the test failures.. 
- some of these look like use of "sh" is getting picked up by agnhost and should be /bin/sh instead
- the sysctl tests are unhappy
- not sure about the cronjob failure

needs rebase

File: test/e2e/instrumentation/logging/generic_soak.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func RunLogPodsWithSleepOf(f *framework.Framework, sleep time.Duration, podname
 			return v1.PodSpec{
 				Containers: []v1.Container{{
 					Name:  "logging-soak",
-					Image: imageutils.GetE2EImage(imageutils.BusyBox),
+					Image: imageutils.GetE2EImage(imageutils.Agnhost),
 					Args: []string{
```
Comment (bclau) [inline test/e2e/instrumentation/logging/generic_soak.go]: Default Agnhost command is ``/agnhost``, so we need to override that with something like ``Command: []string{"/bin/sh"},``

**Comment by bclau** [comment]
> 
> 
> a glance at the test failures..
> 
>     * some of these look like use of "sh" is getting picked up by agnhost and should be /bin/sh instead
> 

Indeed, the default Command of the ``agnhost`` image is ``/agnhost``, so these tests should override that with ``/bin/sh`` or something. Most of the tests which were using ``busybox`` was overriding the Command anyways.

**Comment by saiyan86** [comment]
This PR needs to rebase

**Comment by alejandrox1** [comment]
ping @johnSchnake 
any updates on this?
/milestone v1.17

**Comment by johnSchnake** [comment]
I haven't taken a look at this after my initial try; I need to iterate through the files and try and fixup the known issues that everyone has pointed out here.

**Comment by johnSchnake** [comment]
Just rebased; starting to pick this up again

**Comment by ttousai** [comment]
Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on Nov. 18. Is this PR still intended for 1.17?

**Comment by johnSchnake** [comment]
@ttousai Whenever I try and pick this up it gets pushed back down on my priorities; I'm not particularly optimistic about this being ready for 1.17.

**Comment by ttousai** [comment]
Moving this to 1.18.
/milestone v1.18

**Comment by danwinship** [comment]
needs rebase again

> Whenever I try and pick this up it gets pushed back down on my priorities

breaking it into smaller PRs may help it get merged more easily

**Comment by gianarb** [comment]
Hello @andrewsykim   !
Bug Triage team here for the 1.18 release. This is a friendly reminder that code freeze is scheduled for 5 March. Is this issue still intended for milestone 1.18, or should it extend to the next one? Thanks in advance!

**Comment by gianarb** [comment]
Hello @johnSchnake bug triage team for 1.18 here! Do you think this PR will get merged before code freeze 5 March? Otherwise, I can move it to 1.19

**Comment by gianarb** [comment]
Hello @andrewsykim @johnSchnake @kubernetes/sig-cli-bugs   !
Bug Triage team here for the 1.18 release. This is a friendly reminder that code freeze is scheduled for 5 March. Is this issue still intended for milestone 1.18, or should it extend to the next one? Thanks in advance!

**Comment by gianarb** [comment]
I had a chat with @johnSchnake via Slack and I will move it to 1.19

/milestone v1.19

**Comment by bai** [comment]
👋 Hello from Bug Triage team! This PR has not been updated for a while, so I'd like to check its status. The code freeze is starting June 25th (about 4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.19, is it still planned for this release?

/cc @andrewsykim @johnSchnake @kubernetes/sig-cli-bugs !

**Comment by bai** [comment]
Hey @andrewsykim @johnSchnake @kubernetes/sig-cli-bugs 👋  Code freeze is coming up on July 9th, so I wanted to follow up to see if this one is going to make it into 1.19 🙏 

Or would you prefer bumping it to 1.20?

**Comment by droslean** [comment]
@johnSchnake Are you still working on this PR or we should move it to 1.21?

**Comment by erismaster** [comment]
Hello! I am from the Bug Triage team! This PR has not been updated for a long time, so I'd like to check on the status. The code freeze is starting March 9th, 2021 (about 1 week from now). We want to ensure that each PR has a chance to be merged before code freeze.
As the PRis tagged for 1.21, is it still planned for this release?

**Comment by spiffxp** [comment]
/remove-lifecycle stale
/milestone clear
AFAIK nobody is actively working this. We may be able to close it out, needs review

**Comment by erismaster** [comment]
Thanks @spiffxp!

**Comment by johnSchnake** [comment]
Hey! Just returning after a long absence. I concur, no one had been working on this, it needs major reworking anyways as this was a first round WIP, etc. It makes more sense to close this out and start from scratch if someone wants to pick it up again.

Sorry about it being handled that way.

**Pull request closed without merging**